jadx-gui/src/main/java/jadx/gui/settings/JadxSettingsWindow.java

```diff
@@ -165,6 +193,49 @@ private SettingsGroup makeDeobfuscationGroup() {
 		return deobfGroup;
 	}
 
+	private SettingsGroup makeRenameGroup() {
+		JCheckBox renameCaseSensitive = new JCheckBox();
+		renameCaseSensitive.setSelected(settings.isRenameCaseSensitive());
+		renameCaseSensitive.addItemListener(e -> {
+			settings.setRenameCaseSensitive(e.getStateChange() == ItemEvent.SELECTED);
+			needReload();
+		});
+
+		JCheckBox renameValid = new JCheckBox();
+		renameValid.setSelected(settings.isRenameValid());
+		renameValid.addItemListener(e -> {
+			settings.setRenameValid(e.getStateChange() == ItemEvent.SELECTED);
+			needReload();
+		});
+
+		JCheckBox renamePrintable = new JCheckBox();
+		renamePrintable.setSelected(settings.isRenamePrintable());
+		renamePrintable.addItemListener(e -> {
+			settings.setRenamePrintable(e.getStateChange() == ItemEvent.SELECTED);
+			needReload();
+		});
+
+		SettingsGroup group = new SettingsGroup(NLS.str("preferences.rename"));
+		JLabel renameCaseSensitiveLabel
+			= group.addRow(NLS.str("preferences.rename_case"), renameCaseSensitive);
+		group.addRow(NLS.str("preferences.rename_valid"), renameValid);
+		group.addRow(NLS.str("preferences.rename_printable"), renamePrintable);
+
+		File dir = FileUtils.createTempDir("jadx");
+		if (FileUtils.isCaseSensitiveFS(dir)) {
+			renameCaseSensitive.setSelected(false);
+			renameCaseSensitive.setEnabled(false);
+			renameCaseSensitiveLabel.setEnabled(false);
```

Comment:
Don't do this, leave selection choice to the user. 
Also, this check will not work for masOS, because HFS case sensitivity determined by user and can be different for different drives.